src/HttpDriver/Driver.php

```diff
@@ -62,16 +67,16 @@ public function getUri()
     private function getHttpClient()
     {
         $options = [];
-        if (null !== $this->config->getTimeout()) {
-            $options['timeout'] = $this->config->getTimeout();
+        if ($this->config->hasValue('timeout')) {
+            $options['timeout'] = $this->config->getValue('timeout');
         }
 
-        if (null !== $this->config->getCurlInterface()) {
-            $options['curl'][10062] = $this->config->getCurlInterface();
+        if ($this->config->hasValue('curl_interface')) {
+            $options['curl'][10062] = $this->config->getValue('curl_interface');
         }
 
         if (empty($options)) {
-            return $this->config->getHttpClient();
+            return $this->config->get('http_client');
```

Comment:
If one is using the BaseConfiguration here instead of the HttpConfiguration, then `http_client` MAY be null. Are we fine with that? 

Also, if one is using the HttpConfiguration and set `http_client` to null, should we care?